src/controller.cpp

```diff
@@ -88,23 +83,21 @@ void Controller::setPlan(const nav_msgs::msg::Path & path)
 
 void Controller::setSpeedLimit(const double & speed_limit, const bool & percentage)
 {
-  utils::ControlConstraints constraints;
+  models::ControlConstraints constraints = optimizer_.getDefaultControlConstraints();
 
-  if (speed_limit == nav2_costmap_2d::NO_SPEED_LIMIT) {
-    // Restore default value
-    constraints = utils::ControlConstraints{base_x_vel_, base_y_vel_, base_theta_vel_};
```

Comment:
You need this line, or else if you set the speed limit and then you set it to `NO_SPEED_LIMIT` it will not revert back to the base value.

You removed the check for if its `==` to the no speed limit in order to reset the behavior back to default

Reply:
it gets default values in first lines of the func

Reply:
Oh I see, sorry I missed that line in the diff